### DISCUSSION TITLE
What features should I include for making project-manager plugins easier?

### DISCUSSION THREAD

**Discussion by dugramen** [dugramen/godot-global-project#2] [Ideas]
It's currently pretty difficult to make `project-manager` plugins because node names are random so their paths are unreliable. There's a lot of `get_child(0).get_child(0)` involved.

One feature I'm already planning is a function that takes an array of numbers to greatly reduce the amount of `get_child` in the code. The array would also accept strings, which it would `String.match` on. For example:

```gdscript
node_search(root_node, [0, 0, "*VBoxContainer*", 1, ["*Label*", 1] ])
# Start at the provided root node
# Get the first child
# Get that child's first child
# Of that node's children, find the first whose name string matches "*VBoxContainer*"
# Get that node's 2nd child
# Find the 2nd child to string match "*Label*"
```

Aside from that, I also want to give easy access to special nodes, that people may use frequently. Like the project list container, the top bar where the current inspect button is etc. Which nodes would be desired / used frequently?

**Comment by linkrjr** [comment]
I am new to godot so I am having trouble installing this and making it work.
Can you elaborate on the installation process?

**Reply by dugramen** [reply]
I haven't updated it in years so things may have broke. But it should just be you download this project, and open it in Godot 

To make addons global, add them to this project, and right click the addon folder to change the color (use Orange for addons from the asset library)

For the true global plugins, you'll have to make them yourself as I don't think anyone has actually released any. It's the same as making normal plugins except you put them in the "editor-only" folder and be aware of some limitations I listed in the readme. But if you're new to Godot that's probably not what you're looking for
